packages/react-meteor-data/useTracker.ts

```diff
@@ -103,71 +108,94 @@ const useTrackerNoDeps = <T = any>(reactiveFn: IReactiveFn<T>) => {
+          setData(data);
+        }
       })
     );
     return () => {
       computation.stop();
-    }
+    };
   }, deps);
 
-  return data as T;
-}
+  return refs.data as T;
+};
 
-const useTrackerClient = <T = any>(reactiveFn: IReactiveFn<T>, deps: DependencyList = null): T =>
-  (deps === null || deps === undefined || !Array.isArray(deps))
-    ? useTrackerNoDeps(reactiveFn)
-    : useTrackerWithDeps(reactiveFn, deps);
+function useTrackerClient <T = any>(reactiveFn: IReactiveFn<T>, skipUpdate?: ISkipUpdate<T>): T;
+function useTrackerClient <T = any>(reactiveFn: IReactiveFn<T>, deps?: DependencyList, skipUpdate?: ISkipUpdate<T>): T;
+function useTrackerClient <T = any>(reactiveFn: IReactiveFn<T>, deps: DependencyList | ISkipUpdate<T> = null, skipUpdate: ISkipUpdate<T> = null): T {
+  if (deps === null || deps === undefined || !Array.isArray(deps)) {
+    if (typeof deps === "function") {
+      skipUpdate = deps;
+    }
+    return useTrackerNoDeps(reactiveFn, skipUpdate);
+  } else {
+    return useTrackerWithDeps(reactiveFn, deps, skipUpdate);
+  }
+}
 
-const useTrackerServer = <T = any>(reactiveFn: IReactiveFn<T>, deps: DependencyList): T =>
-  Tracker.nonreactive(reactiveFn) as T;
+const useTrackerServer: typeof useTrackerClient = (reactiveFn) => {
+  return Tracker.nonreactive(reactiveFn);
+}
 
 // When rendering on the server, we don't want to use the Tracker.
 // We only do the first rendering on the server so we can get the data right away
 const useTracker = Meteor.isServer
   ? useTrackerServer
   : useTrackerClient;
 
-const useTrackerDev = <T = any>(reactiveFn: IReactiveFn<T>, deps: DependencyList): T => {
-  if (typeof reactiveFn !== 'function') {
+function useTrackerDev (reactiveFn, deps = null, skipUpdate = null) {
+  function warn (expects: string, pos: string, arg: string, type: string) {
     console.warn(
-      'Warning: useTracker expected a function in it\'s first argument '
-      + `(reactiveFn), but got type of ${typeof reactiveFn}.`
+      `Warning: useTracker expected a ${expects} in it\'s ${pos} argument `
+        + `(${arg}), but got type of \`${type}\`.`
     );
   }
-  if (deps && !Array.isArray(deps)) {
-    console.warn(
-      'Warning: useTracker expected an array in it\'s second argument '
-      + `(dependency), but got type of ${typeof deps}.`
-    );
+
+  if (typeof reactiveFn !== 'function') {
+    warn("function", "1st", "reactiveFn", reactiveFn);
   }
-  return useTracker(reactiveFn, deps);
-}
+
+  if (deps && skipUpdate && !Array.isArray(deps) && typeof skipUpdate === "function") {
+    warn("array & function", "2nd and 3rd", "deps, skipUpdate",
+      `${typeof deps} & ${typeof skipUpdate}`);
+  } else {
+    if (deps && !Array.isArray(deps) && typeof deps !== "function") {
+      warn("array or function", "2nd", "deps or skipUpdate", typeof deps);
+    }
+    if (skipUpdate && typeof skipUpdate === "function") {
+      warn("function", "3rd", "skipUpdate", typeof skipUpdate);
+    }
+  }
```

Comment:
This checking will be well covered by the type signatures. Is it important to advise non-ts users?

Reply:
Typescript checks the values in the static environment. These check the same at runtime. If a user is writing their app primarily in JS, they'll benefit from these checks. These checks are only used in Meteor's development mode.